openedx_tagging/core/tagging/import_export/import_plan.py

```diff
@@ -139,6 +139,13 @@ def generate_actions(
                 tag.external_id: tag for tag in self.taxonomy.tag_set.all()
             }
 
+            for tag in tags:
```

Comment:
@rpenido Maybe to avoid this loop, you can add the delete actions first using the `_build_action()` func, adding a new param `add_first`. What do you think?

Reply:
My first attempt was just to change the `self.actions.append(action)` to `self.actions.insert(0, action)` for the DeleteTag here:
https://github.com/openedx/openedx-learning/blob/707203268e252a05b2d3454b188cd3f628fc15e6/openedx_tagging/core/tagging/import_export/import_plan.py#L64-L65

This didn't solve the problem because we ran the validation below (for the CreateTag action) in the plan stage (before actually running the delete action).
https://github.com/openedx/openedx-learning/blob/707203268e252a05b2d3454b188cd3f628fc15e6/openedx_tagging/core/tagging/import_export/actions.py#L120-L127

I need to skip this validation if the tag value has a DeleteTag action, so I need to build all the delete actions (calling `_build_delete_actions`) before validating the new tags.

I also tried to avoid this loop filtering while we constructed the dict here:
https://github.com/openedx/openedx-learning/blob/707203268e252a05b2d3454b188cd3f628fc15e6/openedx_tagging/core/tagging/import_export/import_plan.py#L138-L140

But because we have tags (from the DB) and tags (from the importing param), the filtering code became nearly intelligible, so I opted for the loop outside (since it is not a nested loop).

Reply:
> My first attempt was just to change the self.actions.append(action) to self.actions.insert(0, action) for the DeleteTag here (this was your suggestion?):

Yes, it is my suggestion.

> But because we have tags (from the DB) and tags (from the importing param), the filtering code became nearly intelligible, so I opted for the loop outside (since it is not a nested loop).

I see, in that case the loop is fine. Thanks for the explanation!